zio-http/src/main/scala/zio/http/codec/HttpCodec.scala

```diff
@@ -75,15 +75,28 @@ sealed trait HttpCodec[-AtomTypes, Value] {
     HttpCodec.Combine[AtomTypes1, AtomTypes1, Value, Value2, combiner.Out](self, that, combiner)
 
   /**
-   * Combines two query codecs into another query codec.
+   * To end the route codec and begin with query codec, and re-interprets as
+   * PathQueryCodec
+   */
+  final def :?[Value2](
+    that: QueryCodec[Value2],
+  )(implicit
+    combiner: Combiner[Value, Value2],
+    ev: HttpCodecType.Path <:< AtomTypes,
```

Comment:
Delete this implicit evidence, instead this function should return: `HttpCodec[AtomTypes with HttpCodecType.Query, combiner.Out]`.

Reply:
I think we shouldn't be able to do `:?` on top of a `QueryCodec`. This is with the following pattern where users need to directly encode a REST query such as:

```scala

GET /api/products?category=electronics&sort=price

```
`/api/products` is `Path`. and we allow `:?` only on top of `Path`